Sources/OpenId4VP/AuthorizationRequest/AuthorizationRequest.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+
+extension Dictionary where Key == String, Value == String {
+    func values(forKeys keys: [String]) -> [String]? {
+        let values = keys.compactMap { self[$0] }
+        return values.count == keys.count ? values : nil
+    }
+}
+
+struct AuthorizationRequest {
+    let clientId: String
+    let presentation_definition: String?
+    let scope: String?
+    let response_type: String
+    let response_mode: String
+    let nonce: String
+    let state: String
+    let response_uri: String
+    
+    static func getAuthorizationRequest(encodedAuthorizationRequest: String, openId4VpInstance: OpenId4VP) throws {
```

Comment:
why we need to pass OpenId4VP instance?

Reply:
We have passed since we need to update the properties of the class.
Added setter methods to update the properties